openTSNE/affinity.py

```diff
@@ -188,8 +205,8 @@ def set_perplexity(self, new_perplexity):
             "Perplexity changed. Recomputing affinity matrix...", self.verbose
         ):
             self.P = joint_probabilities_nn(
-                self.__neighbors[:, :k_neighbors],
-                self.__distances[:, :k_neighbors],
```

Comment:
I think this might be the reason for one of the failing tests. If we change the perplexity from 100 to 30, for instance, then `self.__neighbors` and `self.__distances` will still be N x 300, but we only want to calculate the P matrix for the 90 neighbors.

Reply:
Fixed this.